wordpress-mobile/WordPress-Android
# Enable Gutenberg for all and hide legacy editors options

~**
** WARNING: Don't merge before the 11.8 app freeze has happened (currently scheduled for Monday Feb 11th 2019)
**~

Addresses:
* Part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/336
* https://github.com/wordpress-mobile/gutenberg-mobile/issues/565
* https://github.com/wordpress-mobile/gutenberg-mobile/issues/566
* https://github.com/wordpress-mobile/gutenberg-mobile/issues/567

### Before testing
0. Before using this PR, set the editor to "Visual" in the App settings. This will be useful for testing that Aztec sill be forced where appropriate and the Visual (or the Legacy editor for that matter) won't be used at all.

### To test A:

1. In the App Settings, have the `Use block editor for new posts` to `OFF`
2. Open a new or empty post. Verify that the editor opened is Aztec. Away to verify that is to look for the "H" format bar icon. If there then it's Aztec.
3. Go back to the posts list and open a non-empty post that *doesn't contain* blocks.
4. Verify that it's Aztec again that gets used
5. Go back to the posts list and open a non-empty post *made with Gutenberg before*
6 Verify that the editor used is now Gutenberg. Look for the "undo/redo" buttons in the bottom toolbar, characteristic of Gutenberg at this point.

### To test B:

1. In the App Settings, have the `Use block editor for new posts` to `ON`
2. Open a new or empty post. Verify that the editor opened is Gutenberg. Look for the "undo/redo" buttons in the bottom toolbar, characteristic of Gutenberg at this point.
3. Go back to the posts list and open a non-empty post that *doesn't contain* blocks.
4. Verify that *it's Aztec again* that gets used
5. Go back to the posts list and open a non-empty post *made with Gutenberg before*
6 Verify that the editor used is now Gutenberg. Look for the "undo/redo" buttons in the bottom toolbar, characteristic of Gutenberg at this point.

### To test C:

* Try to build any app flavor (vanilla, zalpha, wasabi) and verify that Gutenberg is available and working.


Update release notes:
